GooglePlayInstant/Editor/QuickDeploy/WwwRequestInProgress.cs

```diff
@@ -0,0 +1,123 @@
+        /// a warning message and stop monitoring a previous request if one was still being monitored.
+        /// </summary>
+        /// <param name="www">A www instance holding a request that was made</param>
+        /// <param name="progressBarTitleText">A descriptive text to display as the title of the progress bar.</param>
+        /// <param name="onResponseAvailableAction">An action to be invoked on www once the response to the request
+        /// is available.</param>
+        public static void TrackProgress(WWW www, string progressBarTitleText, Action<WWW> onResponseAvailableAction)
+        {
+            if (_onGoingRequest != null)
+            {
+                throw new Exception("Cannot start a another request while the previous one is not complete.");
+            }
+
+            _onGoingRequest = new OnGoingRequest(www, progressBarTitleText, onResponseAvailableAction);
+        }
+
+        /// <summary>
+        /// Verifies the state of the currently monitored ongoing request. Displays progress bar for the request if
+        /// it is still going on. If the response is available, the scheduled post-completion action will be invoked on
+        /// the www instance of the ongoing request, the request will stop being monitored and resources used by this
+        /// request will be disposed.
+        /// </summary>
+        public static void Update()
+        {
+            if (_onGoingRequest == null)
+            {
+                return;
+            }
+
+            if (_onGoingRequest.RequestWww.isDone)
+            {
+                EditorUtility.ClearProgressBar();
+                var onGoingRequest = _onGoingRequest;
+                // Set the static field _requestInProgress to null before invoking the  post-completion action
+                // because the action could be executing an action that will read or overwrite the static field.
+                _onGoingRequest = null;
+                onGoingRequest.OnResponseAvailableAction(onGoingRequest.RequestWww);
+                onGoingRequest.Dispose();
+            }
+            else
+            {
+                if (EditorUtility.DisplayCancelableProgressBar(_onGoingRequest.ProgressBarTitleText,
+                    string.Format("Progress: {0}%", Math.Floor(_onGoingRequest.RequestWww.uploadProgress * 100)),
+                    _onGoingRequest.RequestWww.uploadProgress))
+                {
+                    EditorUtility.ClearProgressBar();
+                    _onGoingRequest.Dispose();
+                    _onGoingRequest = null;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Encapsulates a WWW instance used to send an HTTP request, the tItle to display in the progress bar and
+        /// an action to invoke on the WWW instance when the request is complete.
+        /// </summary>
+        private class OnGoingRequest : IDisposable
```

Comment:
Nit: typo OngoingRequest

Reply:
Resolved.
Oh the things you do when english is not your first or second language